### PR TITLE
feat(cli): make `Spinner.message` able to be changed on-the-fly

### PR DESCRIPTION
**Pull request by timreichen**
- makes `Spinner.message` property public

Reason: Allows to change message during active spinner.

Example:
```ts
import { Spinner } from "https://deno.land/std@$STD_VERSION/cli/spinner.ts";
import { delay } from "https://deno.land/std@$STD_VERSION/async/delay.ts";

const spinner = new Spinner( { message: "Loading..." });
spinner.start();
await delay(1000);
spinner.message = "Almost done..."
await delay(1000);
spinner.stop();
```

### CODE REVIEW

**Review by iuioiua — APPROVED** [review]
I like the idea of this. Thank you.

It does make me think that this same change could be applied to some of the other properties.

**Comment by timreichen** [comment]
> I like the idea of this. Thank you.
> 
> It does make me think that this same change could be applied to some of the other properties.

Like `color` and `interval`?

**Comment by iuioiua** [comment]
Yep 👍🏾

**Comment by timreichen** [comment]
> Yep 👍🏾

Should that be part of this PR or a separate one?

**Comment by iuioiua** [comment]
Let's have it part of this one.

**Comment by timreichen** [comment]
Actually, `interval` might need special treatment, else a change won't do anything.

**Comment by timreichen** [comment]
Ok, I updated the code so it works when changing `interval` and `color`.
While at it, I removed the obsolete `#active` prop and made some adjustments for better readability.

File: cli/spinner.ts [hunk 1]
```diff
@@ -63,11 +63,11 @@ export interface SpinnerOptions {
  */
 export class Spinner {
   #spinner: string[];
-  #message: string;
   #interval: number;
+  message: string;
   #color?: Color;
   #intervalId: number | undefined;
-  #active = false;
+  #frameIndex = 0;
```
Comment (iuioiua) [inline cli/spinner.ts:70]: What's the purpose of this change?

**Review by iuioiua — CHANGES_REQUESTED** [review]
Frankly, I don't get some of these changes. I didn't think making color and friends would introduce such logic.

Sorry to change my mind, but can we revert to focusing on making `Spinner.message` public? Sorry for my previous suggestion. We should always keep PRs focused.

File: cli/spinner.ts [hunk 2]
```diff
@@ -43,17 +43,17 @@ export interface SpinnerOptions {
...
-  message?: string;
   /**
    * The time between each frame of the spinner in milliseconds.
    *
-   * @default {75}
+   * @default 75
    */
   interval?: number;
+  /** The message to display next to the spinner. */
+  message?: string;
```
[the remark this answers is not in the document]
Comment (timreichen) [reply cli/spinner.ts:56]: I put internal props `#spinner` and `#interval` on top and changeable props at after.

**Comment by timreichen** [comment]
> Frankly, I don't get some of these changes. I didn't think making color and friends would introduce such logic.
> 
> Sorry to change my mind, but can we revert to focusing on making `Spinner.message` public? Sorry for my previous suggestion. We should always keep PRs focused.

Ok, I reverted the changes and will open other PRs for the `color` and `interval`. There are some catches to be aware of so it works as intended, so the changes are not just to remove the hash.

File: cli/spinner.ts [hunk 3]
```diff
@@ -79,11 +79,51 @@ export class Spinner {
...
+    this.message = message;
+    this.color = color;
+  }
+
+  set color(value: Color | undefined) {
+    this.#color = value ? COLORS[value] : undefined;
+  }
+  get color() {
+    return this.#color;
+  }
```
[the remark this answers is not in the document]
Comment (timreichen) [reply cli/spinner.ts]: because the actual value is determined by `COLORS[value]`

**Review by iuioiua — APPROVED** [review]
Nice! LGTM. Thank you.

**Pull request merged**